OpenDreamShared/Network/Messages/MsgBrowseResource.cs

```diff
@@ -1,26 +1,22 @@
-﻿using System;
-using Lidgren.Network;
+﻿using Lidgren.Network;
 using Robust.Shared.Network;
 using Robust.Shared.Serialization;
 
-namespace OpenDreamShared.Network.Messages {
-    public sealed class MsgBrowseResource : NetMessage {
-        // TODO: Browse should be on its own channel or something.
-        public override MsgGroups MsgGroup => MsgGroups.EntityEvent;
+namespace OpenDreamShared.Network.Messages;
+public sealed class MsgBrowseResource : NetMessage {
+    // TODO: Browse should be on its own channel or something.
+    public override MsgGroups MsgGroup => MsgGroups.EntityEvent;
 
-        public string Filename = String.Empty;
-        public byte[] Data = Array.Empty<byte>();
+    public string Filename = string.Empty;
+    public int DataHash;
 
-        public override void ReadFromBuffer(NetIncomingMessage buffer, IRobustSerializer serializer) {
-            Filename = buffer.ReadString();
-            var bytes = buffer.ReadVariableInt32();
-            Data = buffer.ReadBytes(bytes);
-        }
+    public override void ReadFromBuffer(NetIncomingMessage buffer, IRobustSerializer serializer) {
+        Filename = buffer.ReadString();
+        DataHash = buffer.ReadVariableInt32();
+    }
 
-        public override void WriteToBuffer(NetOutgoingMessage buffer, IRobustSerializer serializer) {
-            buffer.Write(Filename);
-            buffer.WriteVariableInt32(Data.Length);
-            buffer.Write(Data);
-        }
+    public override void WriteToBuffer(NetOutgoingMessage buffer, IRobustSerializer serializer) {
+        buffer.Write(Filename);
+        buffer.WriteVariableInt32(DataHash);
```

Comment:
A data hash feels unlikely to benefit from being encoded as a variable int, this would probably end up taking more space.

Reply:
Once it's changed to be something other than file size at least.

Reply:
might need to wait on RT to make hashing available before we can do this